tensorflow/examples/skflow/lstm_regression.py

```diff
@@ -0,0 +1,169 @@
+    """
+
+    def lstm_cells(layers):
+        if isinstance(layers[0], dict):
+            return [tf.nn.rnn_cell.DropoutWrapper(tf.nn.rnn_cell.BasicLSTMCell(layer['steps'],
+                                                                               state_is_tuple=True),
+                                                  layer['keep_prob'])
+                    if layer.get('keep_prob') else tf.nn.rnn_cell.BasicLSTMCell(layer['steps'],
+                                                                                state_is_tuple=True)
+                    for layer in layers]
+        return [tf.nn.rnn_cell.BasicLSTMCell(steps, state_is_tuple=True) for steps in layers]
+
+    def dnn_layers(input_layers, layers):
+        if layers and isinstance(layers, dict):
+            return learn.ops.dnn(input_layers,
+                                 layers['layers'],
+                                 activation=layers.get('activation'),
+                                 dropout=layers.get('dropout'))
+        elif layers:
+            return learn.ops.dnn(input_layers, layers)
+        else:
+            return input_layers
+
+    def _lstm_model(X, y):
+        stacked_lstm = tf.nn.rnn_cell.MultiRNNCell(lstm_cells(rnn_layers), state_is_tuple=True)
+        x_ = learn.ops.split_squeeze(1, time_steps, X)
+        output, layers = tf.nn.rnn(stacked_lstm, x_, dtype=dtypes.float32)
+        output = dnn_layers(output[-1], dense_layers)
+        return learn.models.linear_regression(output, y)
+
+    return _lstm_model
+
+# predicting sin function
+TIMESTEPS = 5
+RNN_LAYERS = [{'steps': TIMESTEPS}, {'steps': TIMESTEPS, 'keep_prob': 0.5}]
+DENSE_LAYERS = None
+
+regressor = learn.TensorFlowEstimator(model_fn=lstm_model(TIMESTEPS, RNN_LAYERS, DENSE_LAYERS),
+                                      n_classes=0, verbose=1,
+                                      steps=TRAINING_STEPS, optimizer='Adagrad',
+                                      learning_rate=0.03, batch_size=BATCH_SIZE)
+
+X, y = generate_data(np.sin, np.linspace(0, 100, 10000), TIMESTEPS, seperate=False)
+# create a lstm instance and validation monitor
+validation_monitor = learn.monitors.ValidationMonitor(X['val'], y['val'],
+                                                      every_n_steps=PRINT_STEPS,
+                                                      early_stopping_rounds=1000)
+regressor.fit(X['train'], y['train'], monitors=[validation_monitor], logdir=LOG_DIR+'1')
+predicted = regressor.predict(X['test'])
+score = mean_squared_error(predicted, y['test'])
+print ("MSE: %f" % score)
+
+
+# predicting cos-sin function
+TIMESTEPS = 10
+RNN_LAYERS = [{'steps': TIMESTEPS}, {'steps': TIMESTEPS, 'keep_prob': 0.5}]
+DENSE_LAYERS = [2]
+
+regressor = learn.TensorFlowEstimator(model_fn=lstm_model(TIMESTEPS, RNN_LAYERS, DENSE_LAYERS),
+                                      n_classes=0, verbose=1,
+                                      steps=TRAINING_STEPS, optimizer='Adagrad',
```

Comment:
let's move `steps` and `batch_size` params to `fit` call (that's the new API we are slowly replacing the old one to enable better usage). Also can you format something like this:
```
regressor = learn.TensorFlowEstimator(
   model_fn=...., ...
   ...., ....)
```

and for `monitor` / `fit` the same.